chain/x/zoracle/internal/keeper/raw_request.go

```diff
@@ -59,7 +62,12 @@ func (k Keeper) AddNewRawDataRequest(
 		// TODO: fix error later
 		return types.ErrRequestNotFound(types.DefaultCodespace)
 	}
-	k.SetRawDataRequest(ctx, requestID, externalID, types.NewRawDataRequest(dataSourceID, calldata))
+
+	err := k.SetRawDataRequest(ctx, requestID, externalID, types.NewRawDataRequest(dataSourceID, calldata))
```

Comment:
It's my fault, please change `SetRawDataRequest` signature to not return an error and you don't need to check error when `SetRawDataRequest`